### PR TITLE
feat: add platform flag for ec2 image import task

### LINKED ISSUE #433: Amazon import-image task stuck on "updating" when platform is not specified
#### Overview of the Issue

The  post-processor amazon-import (import-image-task) gets stuck in "updating" because the platform is not provided.

The AWS EC2 equivalent of this process provides an optional `--platform` flag which is not available in the `packer-plugin-amazon` plugin.

#### Reproduction Steps

`
   post-processor "amazon-import" {
      access_key          = var.aws_access_key
      secret_key          = var.aws_access_secret_id
      token               = var.aws_session_token
      region              = var.region
      s3_bucket_name      = var.s3_bucket_name
      format              = "vmdk"
      keep_input_artifact = true
      ami_name            = "packer-demo-${formatdate("YYYY-MM-DD'T'hh-mm-ssZ", timestamp())}"
      role_name           = var.role_name
      ami_encrypt         = var.ami_encrypt
      ami_kms_key         = var.ami_kms_key
      ami_users           = var.ami_users
      ami_ou_arns         = var.ami_ou_arns
      boot_mode           = var.boot_mode
      license_type        = var.license_type
    }
`

### Result 
`
        {
            "Architecture": "x86_64",
            "Encrypted": true,
            "ImportTaskId": "import-ami-03cf3969904860ae8",
            "KmsKeyId": "XXXXXXXXXXXXXXXXXXXXX",
            "LicenseType": "BYOL",
            "Progress": "27",
            "SnapshotDetails": [
                {
                    "DeviceName": "/dev/sde",
                    "DiskImageSize": 9525806592.0,
                    "Format": "VMDK",
                    "Status": "completed",
                    "UserBucket": {
                        "S3Bucket": "XXXXXXXXXXXXXXXXXXXXX",
                        "S3Key": "packer-import-1699295410.vmdk"
                    }
                }
            ],
            "Status": "active",
            "StatusMessage": "updating",
            "Tags": [],
            "BootMode": "uefi"
        },
`
Which eventually gets deleted.

### Plugin and Packer version
`
Packer v1.9.4
`
`
 amazon = {
   version = ">=1.2.7"
   source = "github.com/hashicorp/amazon"
}
`

### Operating system and Environment details
`
NAME="Ubuntu"
VERSION="20.04.6 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.6 LTS"
VERSION_ID="20.04"
`
### Log Fragments and crash.log files
`
    vsphere-iso.windows (amazon-import): Completed upload of /home/user/cicd-base-windows10-template/artifacts//windows-10-2H1-pro-main-2023-11-10T01-49-06Z.vmdk to s3://XXXXXXXXXXXXX/packer-import-1699580946.vmdk
    vsphere-iso.windows (amazon-import): Setting license type to 'BYOL'
    vsphere-iso.windows (amazon-import): Started import of s3://XXXXXXXXXXXX/packer-import-1699580946.vmdk, task id import-ami-0b0d543bf1549c983
    vsphere-iso.windows (amazon-import): Waiting for task import-ami-0b0d543bf1549c983 to complete (may take a while)
`
`
        {
            "Architecture": "x86_64",
            "Encrypted": true,
            "ImportTaskId": " import-ami-0b0d543bf1549c983",
            "KmsKeyId": "XXXXXXXXXXXXXXXXXXX",
            "LicenseType": "BYOL",
            "Progress": "27",
            "SnapshotDetails": [
                {
                    "DeviceName": "/dev/sde",
                    "DiskImageSize": 9525806592.0,
                    "Format": "VMDK",
                    "Status": "completed",
                    "UserBucket": {
                        "S3Bucket": "XXXXXXXXXXXXXXXXXX",
                        "S3Key": "packer-import-1699580946.vmdk"
                    }
                }
            ],
            "Status": "active",
            "StatusMessage": "updating",
            "Tags": [],
            "BootMode": "uefi"
        }
`

Alternatively I have tested the AWS CLI Command that does the same thing 
`
    post-processor "shell-local" {
      inline = [ <<EOF
        aws ec2 import-image --role-name "${var.role_name}" \
          --description ${local.vm_name_description} \
          --license-type BYOL \
          --disk-containers Format=vmdk,UserBucket="{S3Bucket=${var.s3_bucket_name},S3Key=${local.vm_name_description}.vmdk}"
        EOF
      ]
    }
`

and got the same results. But setting platform in the CLI command works.
`
    post-processor "shell-local" {
      inline = [ <<EOF
        aws ec2 import-image --role-name "${var.role_name}" \
          --description ${local.vm_name_description} \
          --platform windows \
          --license-type BYOL \
          --disk-containers Format=vmdk,UserBucket="{S3Bucket=${var.s3_bucket_name},S3Key=${local.vm_name_description}.vmdk}"
        EOF
      ]
    }
`

### PR DESCRIPTION
**Pull request by GastroGee**
Setting platform for the image-import task fixes a bug where the process gets stuck as mentioned in #433 

If your PR resolves any open issue(s), please indicate them like this so they will be closed when your PR is merged:

Closes #433

### CODE REVIEW

File: post-processor/import/post-processor.go [hunk 1]
```diff
@@ -146,6 +151,11 @@ func (p *PostProcessor) Configure(raws ...interface{}) error {
 			errs, fmt.Errorf("invalid boot mode '%s'. Only 'uefi' and 'legacy-bios' are allowed", p.config.BootMode))
 	}
 
+	if p.config.Platform != "linux" && p.config.Platform != "windows" {
```
Comment (lbajolet-hashicorp) [inline post-processor/import/post-processor.go:154]: I would suggest accepting the empty string as well, but warn/error if the boot mode is uefi
```suggestion
	switch p.config.Platform {
	case "linux", "windows":
	case "":
		if p.config.BootMode == "uefi" {
			errs = packersdk.MultiErrorAppend(errs, "'platform' must be set for 'uefi' image imports")
		}
	default:
		errs = packersdk.MultiErrorAppend(errs, fmt.Errorf(
			"invalid platform '%s'. Only 'linux' and 'windows' are allowed",
			p.config.Platform))
```

**Review by lbajolet-hashicorp — CHANGES_REQUESTED** [review]
Hi @GastroGee,

Thanks for opening this PR! I left a couple suggestions regarding the default value for the attribute, I propose leaving it empty by default, unless this is required (i.e. if `boot_mode` is set).

Other than that, the implementation makes sense.
Out of curiosity, have you tested this on a template? Does that fix the problem?

Thanks again for addressing this bug!

File: post-processor/import/post-processor.go (hunk 1 above)
Comment (GastroGee) [reply post-processor/import/post-processor.go]: Updated to reflect suggestion. I also tested locally with platform set to windows and platform set to empty (when boot_mode is uefi)

**Review by lbajolet-hashicorp — CHANGES_REQUESTED** [review]
Hi @GastroGee,

The code looks good to me, there's some errors at build-time though, could you fix those and re-submit?

Once that's done, we can merge this, thanks!

**Review by lbajolet-hashicorp — APPROVED** [review]
LGTM, thanks for the contribution @GastroGee!
